src/simplestreams/simple_streams_manifest.cpp

```diff
@@ -115,6 +127,14 @@ std::unique_ptr<mp::SimpleStreamsManifest> mp::SimpleStreamsManifest::fromJson(c
         {
             const auto version_string = it.key();
             const auto version = versions[version_string].toObject();
+            const auto version_from_official = manifest_products_from_official[product_key]
+                                                   .toObject()["versions"]
+                                                   .toObject()[version_string]
+                                                   .toObject();
+
+            if (version != version_from_official)
+                continue;
```

Comment:
Hi @townsend2010 , 

This part of code will make sure the invalid remote info will be ruled out.

Reply:
Hi @rim99,

Ok, I see, thanks for pointing that out.  Are we intending the manifest from the mirror to be **exactly** the same as the manifest from the official host?  If so, then I think this should be more robust in checking, like computing a checksum for the official manifest and then computing the checksum of the mirror manifest and comparing.  Otherwise, if there are differences, I think this should check *everything* possible that would be the same, not just `version`.

Reply:
These are the json file for comparing:
* [manifest from Official](https://cloud-images.ubuntu.com/releases/streams/v1/com.ubuntu.cloud:released:download.json)
* [manifest from a mirror](https://repo.huaweicloud.com/ubuntu-cloud-images/releases/streams/v1/com.ubuntu.cloud:released:download.json)

Initially, I was trying to validate it by comparing whole file. And it doesn't works since , like I mentioned before, the mirror only syncs with official site periodically. There are chances that, a version released at the official site, but doesn't exist at the mirror site. 

So current implementation picks versions from manifest of mirror site, since they are available on the mirror site at the moment. It doesn't compare the info other than version, because these are useless for an image, which is what it want ultimately. The crucial info has been verified.

Reply:
Hi @rim99,

Ok, I follow what you are saying now and I agree with your assessment.  Thanks!